common/src/main/java/com/revenuecat/purchases/common/diagnostics/DiagnosticsFileHelper.kt

```diff
@@ -0,0 +1,42 @@
+package com.revenuecat.purchases.common.diagnostics
+
+import com.revenuecat.purchases.common.FileHelper
+import com.revenuecat.purchases.common.verboseLog
+import org.json.JSONObject
+
+/**
+ * All methods in this file should be executed within the diagnostics thread to ensure there are no threading issues.
+ */
+class DiagnosticsFileHelper(
+    private val fileHelper: FileHelper
+) {
+    companion object {
+        const val DIAGNOSTICS_FILE_PATH = "RevenueCat/diagnostics/diagnostic_events.jsonl"
+    }
+
+    @Synchronized
+    fun appendEventToDiagnosticsFile(diagnosticsEvent: DiagnosticsEvent) {
+        fileHelper.appendToFile(DIAGNOSTICS_FILE_PATH, diagnosticsEvent.toString() + "\n")
+    }
+
+    @Synchronized
+    fun cleanSentDiagnostics(diagnosticsSentCount: Int) {
+        fileHelper.removeFirstLinesFromFile(DIAGNOSTICS_FILE_PATH, diagnosticsSentCount)
+    }
+
+    @Synchronized
+    fun deleteDiagnosticsFile() {
+        if (!fileHelper.deleteFile(DIAGNOSTICS_FILE_PATH)) {
+            verboseLog("Failed to delete diagnostics file.")
```

Comment:
Should this be a warning instead?

Reply:
Yeah, I made all logs for diagnostics `verbose`, to not pollute the normal logs. Lmk if you think we should still send these as warnings/errors though, I'm not opposed to either approach.